code/modules/admin/verbs/secrets.dm

```diff
@@ -348,16 +348,36 @@ GLOBAL_DATUM(everyone_a_traitor, /datum/everyone_is_a_traitor_controller)
 			if(!is_funmin)
 				return
 			SSblackbox.record_feedback("nested tally", "admin_secrets_fun_used", 1, list("Break All Lights"))
-			message_admins("[key_name_admin(holder)] broke all lights")
+			var/station_only = tgui_alert(usr,"Break which lights?", "And the lord said...", list("All Lights", "Station Z Only", "Cancel"))
+			switch(station_only)
+				if("All Lights")
+					station_only = FALSE
+				if("Station Z Only")
+					station_only = TRUE
+				if("Cancel")
+					return
```

Comment:
```suggestion
				else
					return
```

if you get invalid data from the prompt, you must still cancel, incase they hit the [x] in the top corner or disconnect.